ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -631,7 +669,8 @@ private void prepareForSending() {
     void onFillComplete() {
       buf.flip();
       prepareForSending();
-      chunkReadyTimeMs = time.milliseconds();
+      chunkReadyAtMs = time.milliseconds();
+      routerMetrics.chunkFillTimeMs.update(chunkReadyAtMs - chunkFreeAtMs);
```

Comment:
@xiahome good point, these metrics do not make sense for the metadata chunk. will make the fix to update only for data chunks.

@vgkholla This is mentioned in the description. This is the overall time for the chunk to get filled and the claim is that this is primarily going to account for the wait time for data to become available in the channel. We can add more granularity if we wish, but this was a simple and non-cluttering way to add a useful metric.